vertx-web/src/main/java/io/vertx/ext/web/impl/RoutingContextImpl.java

```diff
@@ -77,35 +78,45 @@ public RoutingContextImpl(String mountPoint, RouterImpl router, HttpServerReques
   void route() {
     String path = request.path();
     HostAndPort authority = request.authority();
-    if ((authority == null && request.version() != HttpVersion.HTTP_1_0) || path == null || path.isEmpty()) {
-      // Authority must be present (HTTP/1.x host header // HTTP/2 :authority pseudo header)
-      // HTTP paths must start with a '/'
-      fail(400);
-    } else if (path.charAt(0) != '/') {
-      // For compatiblity we return `Not Found` when a path does not start with `/`
+
+    if (authority == null && request.version() != HttpVersion.HTTP_1_0) {
+      String message = HttpVersion.HTTP_1_1 == request.version() ?
+        "For HTTP/1.x requests, the 'Host' header is required" :
+        "For HTTP/2 requests, the ':authority' pseudo-header is required";
+      fail(400, new IllegalStateException(message));
```

Comment:
I don't think capturing a stacktrace is necessary in this case. In other places, for similar purpose, we use `VertxException`:

https://github.com/vert-x3/vertx-web/blob/100755288d3e2b4268f86959d23e14193dd581db/vertx-web/src/main/java/io/vertx/ext/web/handler/impl/CSRFHandlerImpl.java#L191